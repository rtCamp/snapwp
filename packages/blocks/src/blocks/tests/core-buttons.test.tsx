import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CoreButtons from '../core-buttons';

jest.mock( '@wordpress/style-engine', () => {
	const actual = jest.requireActual( '@wordpress/style-engine' );
	return {
		...actual,
		compileCSS: jest.fn( ( style ) => style ),
	};
} );

describe( 'CoreButtons', () => {
	beforeEach( () => {
		jest.clearAllMocks();
	} );
	test( 'renders with given className and style', () => {
		const attributes = {
			cssClassName: 'test-class',
			style: JSON.stringify( { color: 'red' } ),
		};
		const children = <button>Click Me</button>;

		const { asFragment } = render(
			<CoreButtons attributes={ attributes }>{ children }</CoreButtons>
		);

		const divElement = screen.getByRole( 'button', {
			name: 'Click Me',
		} ).parentElement;
		expect( divElement ).toHaveClass( 'test-class' );
		expect( divElement ).toHaveStyle( 'color: red' );
		expect( asFragment() ).toMatchSnapshot();
	} );

	test( 'renders with default attributes when none are provided', () => {
		const children = <button>Click Me</button>;

		const { asFragment } = render(
			<CoreButtons>{ children }</CoreButtons>
		);

		const divElement = screen.getByRole( 'button', {
			name: 'Click Me',
		} ).parentElement;
		expect( divElement ).not.toHaveClass();
		expect( divElement ).not.toHaveStyle( '' );
		expect( asFragment() ).toMatchSnapshot();
	} );

	test( 'renders children correctly', () => {
		const attributes = {
			cssClassName: 'test-class',
			style: JSON.stringify( { color: 'red' } ),
		};
		const children = (
			<>
				<button>Button 1</button>
				<button>Button 2</button>
			</>
		);

		const { asFragment } = render(
			<CoreButtons attributes={ attributes }>{ children }</CoreButtons>
		);

		expect(
			screen.getByRole( 'button', { name: 'Button 1' } )
		).toBeInTheDocument();
		expect(
			screen.getByRole( 'button', { name: 'Button 2' } )
		).toBeInTheDocument();
		expect( asFragment() ).toMatchSnapshot();
	} );
} );
