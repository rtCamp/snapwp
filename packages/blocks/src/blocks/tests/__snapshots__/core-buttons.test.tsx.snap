// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreButtons renders children correctly 1`] = `
<DocumentFragment>
  <div
    class="test-class"
    style="color: red;"
  >
    <button>
      Button 1
    </button>
    <button>
      Button 2
    </button>
  </div>
</DocumentFragment>
`;

exports[`CoreButtons renders with default attributes when none are provided 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <button>
      Click Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`CoreButtons renders with given className and style 1`] = `
<DocumentFragment>
  <div
    class="test-class"
    style="color: red;"
  >
    <button>
      Click Me
    </button>
  </div>
</DocumentFragment>
`;
